### DISCUSSION TITLE
Is it possible to modify the Service URL subpath?

### DISCUSSION THREAD

**Discussion by bgdnxt** [connectrpc/connect-go#462] [General]
current generated connect service  baseURL like below:

```go

say: connect_go.NewClient[v1.SayRequest, v1.SayResponse](
			httpClient,
			baseURL+"/buf.connect.demo.eliza.v1.ElizaService/Say",
			opts...,
		),

```

could we have a way to modify `/buf.connect.demo.eliza.v1.ElizaService` subpaath?

**Comment by bgdnxt** [comment]
and also the generated  service handler's

**Comment by akshayjshah** [comment]
You can _prefix_ the standard paths, but otherwise you can't modify them. If you look in the generated code, though, you'll see that it's a _very_ thin layer on top of `connect.NewClient` and `connect.NewHandler` - you can use those directly, or you can modify `protoc-gen-connect-go` to generate exactly the code you'd like.

At least right now, I don't want to support arbitrary path modifications in `connect-go`. The path is part of the gRPC and Connect protocols, and freeform modifications make it much harder to generate working clients.
